raft/src/test/java/org/apache/kafka/raft/internals/KafkaRaftMetricsTest.java

```diff
@@ -137,14 +137,14 @@ public void shouldRecordNonVoterQuorumState() throws IOException {
         state.initialize(new OffsetAndEpoch(0L, 0));
         raftMetrics = new KafkaRaftMetrics(metrics, "raft", state);
 
-        assertEquals("unattached", getMetric(metrics, "current-state").metricValue());
+        assertEquals("observer", getMetric(metrics, "current-state").metricValue());
         assertEquals((double) -1L, getMetric(metrics, "current-leader").metricValue());
         assertEquals((double) -1L, getMetric(metrics, "current-vote").metricValue());
         assertEquals((double) 0, getMetric(metrics, "current-epoch").metricValue());
         assertEquals((double) -1L, getMetric(metrics, "high-watermark").metricValue());
 
         state.transitionToFollower(2, 1);
-        assertEquals("follower", getMetric(metrics, "current-state").metricValue());
+        assertEquals("observer", getMetric(metrics, "current-state").metricValue());
```

Comment:
This change makes sense to me. But I'm not quite sure if it's correct that the `unattached` state becomes `observer`. ping @dengziming @jsancio @hachikuji

Reply:
In our code design, observer/voter are the state of follower,  leader/candidate/voted/follower/unattached/Resigned are the state of raft node state machine, so IMO, we'd better add "resigned" state instead of "observer" state.

Reply:
what I was referring here was a broker, which is not part of the controller quorum, but it still fetches the metadata topic from the leader controller but it is not part of the voters, because it has "broker" role and not part of the quorum. Isn't it an "observer"?
This is also what we get by using the `kafka-metadata-quorum` bin tool which shows the brokers in the observer list. The purpose of the PR was kind of aligning the tool output with the KRaft current state metric which was not reporting a broker as "observer" but as "follower".

Reply:
Yes, `resigned` is another issue, in the PR we can try to distinguish  `observer` and `follower` to match the terminology we use in public api.

Reply:
Thanks, @dengziming ! :)